src/secret_storage.ts

```diff
@@ -0,0 +1,29 @@
+import { ExtensionContext, SecretStorage } from "vscode"
+
+export default class AuthSettings {
```

Comment:
We don't need another singleton here, thanks for figuring out how to make this work.
I'll close the PR and open my own to save the back and forth.